dataRetriever/dataPool/sovereign/outGoingOperationPool.go

```diff
@@ -0,0 +1,106 @@
+package sovereign
+
+import (
+	"sort"
+	"sync"
+	"time"
+
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("outgoing-operations-pool")
+
+type cacheEntry struct {
+	data     []byte
+	expireAt time.Time
+}
+
+// This is a cache which stores outgoing txs data at their specified hash.
+// Each entry in cache has an expiry time. We should delete entries from this cache once the confirmation from the notifier
+// is received that the outgoing operation has been sent to main chain.
+// An unconfirmed operation is a tx data operation which has been stored in cache for longer than the time to wait for
+// unconfirmed outgoing operations.
+// The leader of the next round should check if there are any unconfirmed operations and try to resend them.
+
+// TODO: [TBD] here a mechanism to check for unconfirmed operations inside the network as well in order not to double spend?
```

Comment:
Double spend is not possible. as mainchain SC will not let it. TThe thing is that the leader has to check whether there is something in the cache and resend the txData to the mainchain if that was not confirmed. This is the reason for this cache.

Reply:
Yes, but each validator/leader runs its own notifier. Maybe next leader doesn't get notified (his notifier goes down) that the previous leader sent the data and he tries to do it again. Double spend at the sc level is not possible, is just that the next validator might try to resend something that has been already sent

Reply:
he only spends gas. Delete double spend comments.

Reply:
Deleted comment